solaris/solaris11/generate_wazuh_packages.sh

```diff
@@ -208,45 +213,108 @@ clean() {
 }
 
 
-show_help()
-{
-  echo "
-  This scripts build wazuh package for Solaris 11 Intel based architecture.
-  USAGE: Command line options available:
-    -h, --help         Displays this help.
-    -d, --download     Download Wazuh repository.
-    -b, --build        Builds Solaris11 packages.
-    -u, --utils        Download and install utilities and dependencies.
-    -c, --clean-all    Clean sources, local respository and generated files.
-  "
+show_help() {
+  echo
+  echo "Usage: $0 [OPTIONS]"
+  echo
+  echo "    -b, --branch <branch>               Select Git branch or tag e.g. $wazuh_branch"
+  echo "    -e, --environment                   Install all the packages necessaries to build the pkg package"
+  echo "    -s, --store  <pkg_directory>        Directory to store the resulting pkg package. By default: /tmp/build"
+  echo "    -p, --install-path <pkg_home>       Installation path for the package. By default: /var"
+  echo "    -c, --checksum                      Compute the SHA512 checksum of the pkg package."
+  echo "    -h, --help                          Shows this help"
+  echo
+  exit $1
 }
 
-# Reading command line arguments
-key="$1"
-case $key in
-  -h|--help)
-    show_help
-    exit 0
-    ;;
-  -d|--download)
+build_package() {
     download_source
-    exit 0
-    ;;
-  -b|--build)
     create_repo
     compile
     create_package
-    exit 0
-    ;;
-    -u|--utils)
-    utils_and_dependencies
-    exit 0
-    ;;
-    -c|--clean-all)
     clean
     exit 0
-    ;;
-  *)
-esac
+}
+
+# Main function, processes user input
+main() {
+  # If the script is called without arguments
+  # show the help
+  if [[ -z $1 ]] ; then
+    show_help 0
+  fi
+
+  build_env="no"
+  build_pkg="no"
+
+  while [ -n "$1" ]
+  do
+    case $1 in
+        "-b"|"--branch")
+            if [ -n "$2" ]
+            then
+                wazuh_branch="$2"
+                build_pkg="yes"
+                shift 2
+            else
+                show_help 1
+            fi
+        ;;
+        "-h"|"--help")
+            show_help
+            exit 0
+        ;;
+        "-e"|"--environment" )
+            build_environment
+            exit 0
+        ;;
+        "-p"|"--install-path")
+            if [ -n "$2" ]
+            then
+                install_path="$2"
+                shift 2
+            else
+                show_help 1
+            fi
+        ;;
+        "-s"|"--store")
+            if [ -n "$2" ]
+            then
+                target_dir="$2"
+                shift 2
+            else
+                show_help 1
+            fi
+        ;;
+        "-c" | "--checksum")
+            if [ -n "$2" ]; then
+                checksum_dir="$2"
+                compute_checksums="yes"
+                shift 2
+            else
```

Comment:
Same, what happens if `-c` is empty?

Reply:
Done: https://github.com/wazuh/wazuh-packages/commit/e1d2b0f75c918f86466d58539581845dbb7a3b75